wallet/src/actors/worker/methods.rs

```diff
@@ -1153,3 +1197,18 @@ impl Worker {
         wallet.export_master_key(password).map_err(Error::from)
     }
 }
+
+fn validate_birth_date(
+    birth_date: u32,
+    checkpoint_zero_timestamp: i64,
+    checkpoints_period: u16,
+) -> Result<()> {
+    let seconds_from_genesis = get_timestamp() - checkpoint_zero_timestamp;
+    let current_epoch = seconds_from_genesis / i64::from(checkpoints_period);
+
+    if current_epoch < i64::from(birth_date) {
+        Err(Error::InvalidBirthDate(birth_date, current_epoch as u32))
```

Comment:
You can use this:

```suggestion
        Err(Error::InvalidBirthDate(birth_date, u32::try_from(current_epoch).unwrap_or(0)))
```

If for some reason the current epoch is negative, then the error message will show a 0. And if the current epoch is greater than u32::MAX, then it will not enter this branch so we don't care.